lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -413,6 +414,39 @@ getLocationInfoAsString(const SmallPtrSet<Operation *, 8> &ops) {
   return sstr.str();
 }
 
+/// Emit an optional hardware comment that may exist as an attribute on an
+/// operation.  Return true if a comment was found (and something was written to
+/// the output stream) and false otherwise.
+static bool emitComment(raw_ostream &os, Operation *op) {
+  auto comment =
+      op->getAttrOfType<hw::CommentAttr>(hw::CommentAttr::getMnemonic());
+  if (!comment)
+    return false;
+
+  os << "// " << comment.getText().getValue() << "\n";
+  return true;
+}
+
+static bool emitAttribute(raw_ostream &os, Operation *op) {
+  auto attribute = op->getAttrOfType<ArrayAttr>("attribute_instances");
```

Comment:
Plz make these be declared attributes on the ops that support them.

Reply:
Much better design point. Thanks. I'll update this to use this as opposed to speculatively digging around for attributes.

Reply:
I'm going to dig into this on a separate branch. To give this the same treatment as the comment on a HWModuleOp I need to figure out how to deal with HW needing an SV-defined attribute and the circular dependency that seems to describe. 🤔